src/libs/actions/Policy.js

```diff
@@ -30,7 +61,74 @@ function getPolicySummaries() {
         });
 }
 
+/**
+ * Fetches the policyList from the API and saves a simplified version in Onyx
+ */
+function getPolicyList() {
+    GetPolicyList()
+        .then((data) => {
+            if (data.jsonCode === 200) {
+                const policyDataToStore = _.reduce(data.policyList, (memo, policy) => ({
+                    ...memo,
+                    [`${ONYXKEYS.COLLECTION.POLICY}${policy.id}`]: getSimplifiedMemberList(policy),
+                }), {});
+                Onyx.mergeCollection(ONYXKEYS.COLLECTION.POLICY, policyDataToStore);
+            }
+        });
+}
+
+/**
+ * Merges the passed in login into the specified policy
+ *
+ * @param {String} login
+ * @param {String} welcomeNote
+ * @param {String} policyID
+ */
+function invite(login, welcomeNote, policyID) {
+    // Optimistically add the user to the policy
+    const key = `${ONYXKEYS.COLLECTION.POLICY}${policyID}`;
+    const dataToStore = {};
+    dataToStore[key] = {
+        employeeList: allPolicies[key].employeeList.concat([login]),
+    };
+
+    Onyx.mergeCollection(ONYXKEYS.COLLECTION.POLICY, dataToStore)
+        .then(() => {
+            const policyDataWithoutLogin = {};
+            policyDataWithoutLogin[key] = {
+                employeeList: _.without(allPolicies[key].employeeList, login),
+            };
+            Onyx.mergeCollection(ONYXKEYS.COLLECTION.POLICY, policyDataWithoutLogin)
+                .then(() => {
+                   console.debug('done');
+                });
+        });
```

Comment:
I find merge useful if you only have partial data for a given object - delta, and want to merge it to the object without having to retrieve it from storage first, but since you already have the full data in `allPolicies` you can use `set` to write exactly what you want

```js
    const policy = _.clone(allPolicies[key]); // make a shallow copy to preserve original data
    policy.employeeList = [...policy.employeeList, login] // concat the login

    // use set
    Onyx.set(key, policy)
        .then(() => {
            const policyDataWithoutLogin = _.clone(allPolicies[key]);
            policyDataWithoutLogin.employeeList = _.without(allPolicies[key].employeeList, login),

             Onyx.set(key, policyDataWithoutLogin )
                .then(() => {
                   console.debug('done');
                });
        });
```

An slightly shorter alternative

```js
    const policy = _.clone(allPolicies[key]);
    const policyWithLogin = { ...policy, employeeList: [...policy.employeeList, login] };

    Onyx.set(key, policyWithLogin)
        .then(() => {
             // policy is the original policy without the login and we can just set it again
             Onyx.set(key, policy)
                .then(() => {
                   console.debug('done');
                });
        });
```

Reply:
`mergeCollection` would not remove items from an array, it can only overwrite or add some, but it does not work as concat

Reply:
Awesome, thanks for having a look! I'll plan on using set then 👍